pkg/asset/installconfig/aws/validation.go

```diff
@@ -329,12 +330,28 @@ func validateEndpointAccessibility(endpointURL string) error {
 	if port == "" {
 		port = "https"
 	}
-	conn, err := proxy.Dial(context.Background(), "tcp", net.JoinHostPort(URL.Hostname(), port))
-	if err != nil {
-		return err
+	err = proxyDialURL(context.Background(), net.JoinHostPort(URL.Hostname(), port))
+	return err
+}
+
+func proxyDialURL(ctx context.Context, URL string) error {
+	conn, err := proxy.Dial(context.Background(), "tcp", URL)
+	if err == nil {
+		conn.Close()
+		return nil
+	}
+	// DefaultClient picks the http and https proxy environment variables.
+	req, err := http.NewRequestWithContext(context.Background(), "CONNECT", URL, nil)
+	if err == nil {
+		resp, err := http.DefaultClient.Do(req)
```

Comment:
Do we care what the response is? Or does any response indicate we can connect so we are good to go?

Reply:
Don't we have to check the connection for 404 Not Found?

Reply:
> Don't we have to check the connection for 404 Not Found?

Well, see: https://github.com/openshift/installer/pull/5743#discussion_r834342174

I think we're testing whether the endpoint is accessible, and a 404 response from the endpoint indicates our request got a response from the server. If there is a way to get a known response (e.g. 200) from all the endpoints, perhaps that would be a better option, but that is certainly not what was happening prior to these code changes.